File: javascript/array-hasing/contains-duplicate.js
//Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

function duplicate(arr) {
    const nums = new Set()

    for(let num of arr) {
        if (nums.has(num)) {
            return true
        } else {
            nums.add(num)
        }
    }

    return false
}


//testing

console.log(duplicate([2,3,4,5,2]))

File: python/array-hashing/contains-duplicate.py
# Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

def duplicate(arr: list) -> bool:
    nums = set() #hashset
    
    for n in arr:
        if n in nums:
            return True
        nums.add(n)
    return False

#testing
print(duplicate([1,2,3,2]))